tests/integration/API/JSON/v1/FormsTest.php

```diff
@@ -186,40 +192,41 @@ public function testItCanExtractListsFromNestedListSelectionBlock() {
   }
 
   public function testItCanRestoreAForm() {
-    $this->form1->trash();
+    $this->form1->setDeletedAt(new \DateTime());
+    $this->formsRepository->flush();
 
-    $trashedForm = Form::findOne($this->form1->id);
+    $trashedForm = Form::findOne($this->form1->getId());
```

Comment:
Could you please also replace this usage of the old model? In Doctrine you can refresh an entity from DB (`$entityManager->refresh($entity)`).

In this case you could just call `$this->entityManager->refresh($this->form1)` and then make asserts on `$this->form1`.

Reply:
I will replace the usage of the old model as you suggested. I wasn't sure how much time I should spend on doing this for this test class.

On a related note, what are your thoughts on using \PHPUnit\Framework\Assert::assertInstanceOf() instead of assert() for assertions like this one (https://github.com/mailpoet/mailpoet/pull/3401/files#diff-e72a2e2b6300aa942cd56ac3d1cb382dee3064052c53d2aba74b20fc2aaf1f27R199)? I personally prefer more specialized assertions instead of the generic assert(). One of the reasons for that is that we get specific error messages that can help when debugging problems in the test.

Reply:
We use assert because of PHPStan. I think it doesn't understand assertInstanceOf.

Reply:
I was not able to reproduce this but maybe I'm missing something. Anyway, generally speaking, whenever possible I believe we should try to use more specific assertions. So if it indeed PHPStan complains about the PHPUnit assertions maybe we should look into updating our PHPStan configuration file to fix this?

Here is what I did to test it:

```shell
$ git diff tests/integration/API/JSON/v1/FormsTest.php
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
modified: tests/integration/API/JSON/v1/FormsTest.php
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
@ tests/integration/API/JSON/v1/FormsTest.php:199 @ class FormsTest extends \MailPoetTest {
    $this->formsRepository->flush();
    $this->entityManager->refresh($this->form1);

-   assert($this->form1 instanceof FormEntity);
+   $this->assertInstanceOf(FormEntity::class, $this->form1);
    expect($this->form1)->notNull();

    $response = $this->endpoint->restore(['id' => $this->form1->getId()]);

$ php -d memory_limit=-1 /home/rodrigo/devel/mailpoet-dev-env/mailpoet/tasks/phpstan/vendor/bin/phpstan  analyse --configuration=tasks/phpstan/phpstan-tests.neon tests/integration/API/JSON/v1/FormsTest.php
 1/1 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%


                                                                                                                        
 [OK] No errors                                                                                                         
```                                                                                                                        

Should I do something different to see that PHPStan error when we use assertInstanceOf()?

Reply:
@rodrigoprimo Thanks for checking! Sorry, my bad! I think that its `expect($something)->isInstanceOf('Something');` what was causing the issue. Looks like `$this->assertInstanceOf()` works correctly with PHPStan so I guess we should use it. Good find!

Reply:
Thanks for confirming!